firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -321,12 +313,27 @@ public void deleteFirebaseInstallation(
 
         throw new FirebaseInstallationsException(
             "Bad config while trying to delete FID", Status.BAD_CONFIG);
+      } catch (IOException ignored) {
+        retryCount++;
       } finally {
         httpURLConnection.disconnect();
       }
     }
 
-    throw new IOException();
+    throw new FirebaseInstallationsException(
+        "Firebase Installations Service is unavailable. Please try again later.",
+        Status.UNAVAILABLE);
+  }
+
+  private URL getURL(String resourceName) throws FirebaseInstallationsException {
```

Comment:
+1 for adding this method!
An idea for a different name (feel free to ignore):
#getFullyQualifiedRequestUri
#getFullyQualifiedApiEndpoint